rdmd.d

```diff
@@ -470,46 +478,58 @@ private int rebuild(string root, string fullExe,
         }
     }
 
-    auto fullExeTemp = fullExe ~ ".tmp";
+    immutable fullExeTemp = fullExe ~ ".tmp";
+    immutable rootObj = buildPath(objDir, "rdmd.root"~objExt);
+    immutable depsObj = buildPath(objDir, "rdmd.deps"~objExt);
 
-    string[] buildTodo()
+    assert(dryRun || std.file.exists(rootObj),
+        "should have been created by compileRootAndGetDeps");
+
+    int result = 0;
+    string[] objs = [ rootObj ];
+
+    // compile dependencies
+    if (myDeps.byValue.any) // if there is any source dependency at all
```

Comment:
`myDeps` is a `string[string]`. `.byValue.any` is false when there are values but they're all null. Null values are set for non-source dependencies ([lines 656-683](https://github.com/aG0aep6G/tools/blob/10ca1582c4f037a63a1a91befc458458c1e668b2/rdmd.d#L656-L683)).

But maybe this should check against emptiness instead. It's easy to forget that null has a different truthiness than empty strings.